tests/generic_config_updater/test_vlan_interface.py

```diff
@@ -0,0 +1,275 @@
+import logging
+import pytest
+
+from tests.generic_config_updater.gu_utils import apply_patch, expect_op_success, expect_op_failure
+from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+from tests.generic_config_updater.gu_utils import create_checkpoint, delete_checkpoint, rollback_or_reload
+from tests.generic_config_updater.gu_utils import check_show_ip_intf
+# Test on t0 topo to verify functionality and to choose predefined variable
+# "VLAN_INTERFACE": {
+#     "Vlan1000": {},
+#     "Vlan1000|192.168.0.1/21": {},
+#     "Vlan1000|fc02:1000::1/64": {}
+# }
+pytestmark = [
+    pytest.mark.topology('t0'),
+]
+
+logger = logging.getLogger(__name__)
+
+@pytest.fixture(autouse=True)
+def cleanup_test_env(duthosts, rand_one_dut_hostname):
+    """
+    Setup/teardown fixture for VLAN interface config
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    create_checkpoint(duthost)
+
+    yield
+
+    try:
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+        check_show_ip_intf(duthost, "Vlan1000", ["192.168.0.1/21"], [], is_ipv4=True)
+        check_show_ip_intf(duthost, "Vlan1000", ["fc02:1000::1/64"], [], is_ipv4=False)
+    finally:
+        delete_checkpoint(duthost)
+
+def test_vlan_interface_tc1_add_duplicate(duthost):
+    """ Add duplicate v4 and v6 lo intf to config
+
+    Sample output
+        "VLAN_INTERFACE": {
+        "Vlan1000": {},
+        "Vlan1000|192.168.0.1/21": {},
+        "Vlan1000|fc02:1000::1/64": {}
+    }
+    """
+    json_patch = [
+        {
+            "op": "add",
+            "path": "/VLAN_INTERFACE/Vlan1000|192.168.0.1~121",
```

Comment:
>~1 [](http://example.com/codeflow?start=57&length=2)

The `~1` is not easy to read. Could you add a help function to manipulated a human readable string?

Like `join("/VLAN_INTERFACE/", "Vlan1000|192.168.0.1/21")`.

Reply:
I find it is already implemented in library, check [https://python-json-pointer.readthedocs.io/en/latest/mod-jsonpointer.html#jsonpointer.JsonPointer.from_parts](https://python-json-pointer.readthedocs.io/en/latest/mod-jsonpointer.html#jsonpointer.JsonPointer.from_parts)

Reply:
Thanks for the reference. I notice that but didn't realize I could bring it to our code.